index.js

```diff
@@ -325,6 +326,13 @@ Primus.readable('initialise', function initialise(Transformer, options) {
   this.use('authorization', require('./middleware/authorization'));
 
   //
+  // Set the heartbeat interval.
+  //
+  if (options.timeout) {
+    this.interval = setInterval(this.heartbeat.bind(this), options.timeout);
```

Comment:
What I don't like about this is that the timer runs also when there are 0 connections.